apiClients/clientWithTokenApi.ts

```diff
@@ -25,10 +15,16 @@ const clientWithTokenApi = axios.create({
 });
 
 clientWithTokenApi.interceptors.request.use(
-  (config) => {
+  async (config) => {
     const token = cookies().get("token")?.value;
     if (token) {
-      config.headers["Authorization"] = `Bearer ${token}`;
+      const isTokenExpired = validateTokenExpiry(token);
+      if (!isTokenExpired) {
+        config.headers["authorization"] = `Bearer ${token}`;
+      } else {
+        const refreshedToken = await refreshToken();

```

Comment:
Brakuje tutaj zabezpieczenia na wypadek błędu funkcji refreshToken, może to się nie uda i co wtedy? Powinniśmy przekierować usera do logowania lub coś tego rodzaju.